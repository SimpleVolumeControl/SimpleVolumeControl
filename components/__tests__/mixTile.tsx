import { render } from '@testing-library/preact';
import MixTile from '../mixTile';

describe('MixTile', () => {
  [
    'green',
    'lightblue',
    'blue',
    'pink',
    'red',
    'yellow',
    'white',
    'foobar',
  ].forEach((color) =>
    test(`should render correctly (${color})`, () => {
      const { asFragment } = render(
        <MixTile id="mix01" name="Mix 1" color={color} />,
      );
      expect(asFragment()).toMatchSnapshot();
    }),
  );
});
