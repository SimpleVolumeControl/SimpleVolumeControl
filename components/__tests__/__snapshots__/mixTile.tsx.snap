// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MixTile should render correctly (blue) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-blue-500"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-blue-500/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (foobar) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-base-300"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-base-300/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (green) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-lime-500"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-lime-500/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (lightblue) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-cyan-400"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-cyan-400/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (pink) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-pink-500"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-pink-500/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (red) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-red-500"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-red-500/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (white) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-base-300"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-base-300/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`MixTile should render correctly (yellow) 1`] = `
<DocumentFragment>
  <div
    class="card shadow bg-base-100 border-2 border-yellow-400"
  >
    <a
      href="/mix/mix01"
    >
      <div
        class="card-body bg-yellow-400/20"
      >
        <h2
          class="card-title m-0 justify-center"
        >
          Mix 1
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
